connectors/slack/template/index.js

```diff
@@ -1,24 +1,98 @@
+const Mock = require('mock-http');
+const Express = require('express');
+const Url = require('url'); 
+const app = require('./app');
 const configure = require('./configure');
-const Sdk = require('@fusebit/add-on-sdk');
+const Sdk = require('@fusebit/add-on-sdk'); 
 
-/*
+// Express.response is a prototype of the Express response object that Express explicitly sets
+// as the prototype of the Node.js response object passed into the Express API (using Object.setPrototypeOf()).
+// This is how Express injects its APIs over plain Node.js response object. The prototype of 
+// the Express prototype object is Node.js's ServerResponse. Since we want to replace it with a mocked response, 
+// we are going to manually change the prototype of the Express prototype to be the prototype of the mocked 
+// response object. The original inheritance hierarchy was:
+// (Express response) -> ServerResponse -> OutgoingMessage -> Writable -> EventEmitter -> Object
+// After the change, it will be:
+// (Express response) -> (Mocked response) -> Writable -> EventEmitter -> Object
+// And then the request gets similar treatment.
+Object.setPrototypeOf(Object.getPrototypeOf(Express.response), Mock.Response.prototype);
+Object.setPrototypeOf(Object.getPrototypeOf(Express.request), Mock.Request.prototype);
 
-Add-on Settings Manager contract: 
+/**
+* @param ctx {FusebitContext}
+*/
+module.exports = (ctx, cb) => {
+    let url = Url.parse(ctx.url);
+    let pathname = url.pathname.split("/");
+    pathname.splice(0, 5);
+    pathname = "/" + pathname.join("/");
+    url.pathname = pathname;
+    url = Url.format(url);
```

Comment:
There's some magic that's going on here - what is it doing?

Reply:
@tjanczuk is the author of this snippet, and maybe he can explain the magic... I can add a code comment for posterity

Reply:
This looks strongly like something the baseUrl might solve - what's the end result look like?

Reply:
`pathname` ends up with the path after the function which we use to route off of. It took a while to get this right, so I'd rather not tinker with it right now